sdk/search/Azure.Search/src/Options/SearchOptions.cs

```diff
@@ -0,0 +1,145 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using Azure.Core;
+
+namespace Azure.Search
+{
+    /// <summary>
+    /// Options for <see cref="SearchIndexClient.SearchAsync"/> that
+    /// allow specifying filtering, sorting, faceting, paging, and other search
+    /// query behaviors.
+    /// </summary>
+    [CodeGenSchema("SearchRequest")]
+    public partial class SearchOptions : SearchRequestOptions
+    {
+        /// <summary>
+        /// A full-text search query expression;  Use "*" or omit this
+        /// parameter to match all documents.
+        /// </summary>
+        [CodeGenSchemaMember("search")]
+        internal string SearchText { get; set; }
+
+        /// <summary>
+        /// The OData $filter expression to apply to the search query.  You can
+        /// use <see cref="SearchFilter.Create(FormattableString)"/> to help
+        /// construct the filter expression.
+        /// </summary>
+        [CodeGenSchemaMember("filter")]
+        public string Filter { get; set; }
+
+        /// <summary>
+        /// The list of field names to use for hit highlights.  Only searchable
+        /// fields can be used for hit highlighting.
+        /// </summary>
+        public IList<string> HighlightFields { get; set; } = new List<string>();
```

Comment:
Collection properties must not be settable. Devs can still use the collection initializer pattern like we do in KV.

Reply:
Plenty of other instances I won't comment on.